content/terraform/v1.14.x (alpha)/docs/language/import/bulk.mdx

```diff
@@ -0,0 +1,118 @@
+- `locals`: Add the `locals` block to your query file to define temporary variables scoped to the query configuration. Refer to the [`locals` block reference](/terraform/language/block/locals) for configuration details.
+
+### Specify a custom provider configuration
+
+The query file checks the `main.tf` configuration for provider configurations, but you can also declare one or more `provider` blocks in the query configuration file to define alternate provider configurations. To use one of the alternate provider configurations, add the [`provider` argument](#provider) to your `list` block and reference the name of the provider configuration.  Refer to the [`provider` block reference](/terraform/language/block/provider) for information about how to configure `provider` blocks.
+
+## Query the infrastructure
+
+Run the [`terraform query` command](/terraform/cli/commands/query) to retrieve the resource types specified in your `list` blocks. The command prints the results to your console. By default, each result includes the following information:
+
+- Reference to the `list` block that queried for resources formatted as `list.<type>.<label>`.
+- Identity of a discovered resource.   
+
+The provider may also include other information, such as a description of the resource. Refer to your provider documentation for details.
+
+The number of results depends on the provider, as well as any query constraints you configured in the query configuration file.
+
+To generate machine-readable results, you can include the `-json` flag:
+
+```shell-session
+$ terraform query -json
+```
+
+## Generate configuration for importing results
+
+To generate configuration, run the [`terraform query` command](/terraform/cli/commands/query) and add the `-generate-config-out` flag. The flag expects a path where Terraform stores the generated configuration as a file named `generated.tf`. The `generated.tf` file contains the `resource` and `import` blocks, including resource identities, necessary for importing results.
+
+The following example generates configuration to the working directory:
+
+```shell-session
+$ terraform query -generate-config-out=generated.tf
+```
+ 
+After generating the results file, you must remove the file before rerunning the command to generate a new results file. Rerunning the command when a `generated.tf` already exists at the specified path results in an error. 
+
+## Import resources
+
+Copy the `import` and `resource` blocks from the `generated.tf` file to your `main.tf` configuration and run the [`terraform apply` command](/terraform/cli/commands/apply) to import the resources.  
+  
+## Next steps
+
+After importing your resources, you can discard the query configuration file and the `generated.tf` file, but we recommend keeping them as a historical record.
```

Comment:
Copy? Or move? I would assume move otherwise you'd have two couples of these blocks but I'm not sure if there's a special case for the `generated.tf` file

Reply:
Copying is probably what you want to do. I updated the next steps, though, because it's not clear that we meant that you should apply some process to convert the generated configuration into a historical artifact.